filament/backend/src/vulkan/VulkanUtility.h

```diff
@@ -88,10 +88,18 @@ utils::FixedCapacityVector<OutType> enumerate(
 #undef EXPAND_ENUM_NO_ARGS
 #undef EXPAND_ENUM_ARGS
 
-
 // Useful shorthands
 using VkFormatList = utils::FixedCapacityVector<VkFormat>;
 
+// Helpful fixed size array
+template<typename T, size_t capacity>
+class FillArray : public std::vector<T> {
```

Comment:
In general, we'd want to avoid inheriting from std containers because of the slicing issue. Would FixedCapacityVector work instead?

Reply:
FixedCapacityV might work with a bit of work, but what's the slicing issue you're referring to?  I won't be slicing the data in this array.

Reply:
https://en.wikipedia.org/wiki/Object_slicing
In addition, std containers don't have virtual destructors, so it could potentially cause memory leaks for resources in subclasses.
Although I don't see both issues in your class for now, I'd rather avoid anti patterns. For example, someone may pick up your class and want to add more features on it.

Reply:
ok, sounds good.  I've refactored an existing array with similar intent for re-use.  PTAL.  thanks.